reader/src/test/java/org/kaazing/monitoring/reader/CollectorFactoryTest.java

```diff
@@ -18,33 +18,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package org.kaazing.monitoring.reader;
 
-import org.kaazing.monitoring.reader.api.MetricsCollector;
-import org.kaazing.monitoring.reader.impl.MetricsCollectorAgrona;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-/**
- * Factory class that returns a MetricsCollector instance
- */
-public class MetricsCollectorFactory {
+import org.junit.Test;
+import org.kaazing.monitoring.reader.exception.MetricsReaderException;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(MetricsCollectorFactory.class);
+public class CollectorFactoryTest {
 
-    /**
-     * Initializes a MetricCollector instance
-     * @return MetricsCollectorAgrona
-     */
-    public static MetricsCollector getInstance() {
-        Configuration config = new Configuration();
-
-        if (!config.loadConfigFile()) {
-            LOGGER.error("There was a problem with the configuration file. Exiting application.");
-            return null;
+    @Test
+    public void testCollectorFactory() throws Exception {
```

Comment:
We could perform assertions on getMetricsCollector() and getMessagesCollector() manager return objects, but for this it would be required to have the FileProcessor injected (passed via constructor) in order to be able to mock the fileProcessor.getMappedFile() method. However, since the FileProcessor creation is a bit complex and due to the fact that it needs the reader configuration data, I think it would be ok for now to keep the current implementation and keep this logic within the initialize method, even if this makes some of the behavior of this class not testable.